language/move-ir/types/src/ast.rs

```diff
@@ -555,6 +594,75 @@ pub enum Exp_ {
 /// The type for a `Exp_` and its location
 pub type Exp = Spanned<Exp_>;
 
+//**************************************************************************************************
+// Bytecode
+//**************************************************************************************************
+
+pub type BytecodeBlocks = Vec<(Label, BytecodeBlock)>;
+pub type BytecodeBlock = Vec<Bytecode>;
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct Label(pub String);
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Bytecode_ {
+    Pop,
+    Ret,
+    BrTrue(Label),
+    BrFalse(Label),
+    Branch(Label),
+    LdU8(u8),
+    LdU64(u64),
+    LdU128(u128),
+    CastU8,
+    CastU64,
+    CastU128,
+    LdByteArray(ByteArray),
+    LdAddr(AccountAddress),
+    LdTrue,
+    LdFalse,
+    CopyLoc(Var),
+    MoveLoc(Var),
+    StLoc(Var),
+    Call(ModuleName, FunctionName, Vec<Type>),
```

Comment:
Right now we have the module alias here. Do we want to switch to a fully qualified name (addr + module name) in the future? I feel like that will be more suitable as a target of the source compiler as it has already gone through naming and expansion.

Otherwise this enum looks good to me.

Reply:
We probably always plan on having aliases here (as its what we want probably for the IR Bytecode syntax). Recreating a fake alias for the source language is trivial